25.1/xenial/Dockerfile

```diff
@@ -0,0 +1,76 @@
+FROM ubuntu:16.04
+
+# Install dependencies
+RUN apt-get update && \
+    apt-get install -y \
+            autoconf \
+            automake \
+            build-essential \
+            curl \
+            git \
+            imagemagick \
+            ispell \
+            libdbus-1-dev \
+            libgif-dev \
+            libgnutls-dev \
+            libgtk2.0-dev \
+            libjpeg-dev \
+            libmagick++-dev \
+            libncurses-dev \
+            libpng-dev \
+            libtiff-dev \
+            libx11-dev \
+            libxpm-dev \
+            texinfo \
+            && \
+    rm -rf /var/lib/apt/lists/*
+
+# Build emacs
+COPY emacs-25.1 /tmp/emacs/
+RUN cd /tmp/emacs && \
+    ./autogen.sh && \
+    ./configure --prefix=/opt/emacs && \
+    make -j 8 install && \
+    rm -rf /tmp/emacs
+
+#--------------------------------------------------------------------------------
+
+FROM ubuntu:16.04
+
+# Runtime dependencies
+RUN apt-get update && \
+    apt-get install -y \
+            autoconf \
+            automake \
+            build-essential \
+            curl \
+            git \
+            imagemagick \
+            ispell \
+            libdbus-1-dev \
+            libgif-dev \
+            libgnutls-dev \
+            libgtk2.0-dev \
+            libjpeg-dev \
+            libmagick++-dev \
+            libncurses-dev \
+            libpng-dev \
+            libtiff-dev \
+            libx11-dev \
+            libxpm-dev \
+            python \
+            texinfo \
+            && \
+    rm -rf /var/lib/apt/lists/*
+
+# Setup environment variables
+ENV EMACS_BRANCH="emacs-25.1"
+ENV EMACS_VERSION="25.1"
+
+COPY --from=0 /opt/emacs /opt/emacs/
+
+# Install Cask
+ENV PATH="/opt/emacs/bin:/root/.cask/bin:${PATH}"
+RUN curl -fsSL https://raw.githubusercontent.com/cask/cask/master/go | python
```

Comment:
I'd like to see a variant without Cask and python for people like me who don't want or need them in the image.

Reply:
That'd make 5 variants! :-)

I mean, why not but it feels a bit too much, especially since python/cask does not really add much MB or hurt when present.

Reply:
I’ve seen projects separate the build images and the “run” images into separate namespaces so that the naming isn’t so cluttered. 

So for instance most users would never be concerned with the dev images:

    Silex/docker-emacs-builder:25.3 (this is the “dev” image)

But they do select either the emacs-only images

    Silex/docker-emacs:25.3 (this is “slim”)
    Silex/docker-emacs:25.3-alpine (“slim+alpine”)

Or the ones with cask too

    Silex/docker-emacs-cask:25.3 (this is “slim+cask”)
    Silex/docker-emacs-cask:25.3-alpine (“slim+alpine+cask”)


You might find the Alpine images are small enough that adding python is significant.

Reply:
Again I like your idea! It makes sense. I'll start by creating all the images in this repo and then split the repo in 3 when the time is right.